pipenv/vendor/pythonfinder/models/python.py

```diff
@@ -302,6 +317,39 @@ def which(self, name) -> PathEntry | None:
         non_empty_match = next(iter(m for m in matches if m is not None), None)
         return non_empty_match
 
+    def find_python_versions_from_windows_launcher(self):
+        # Open the registry key for Python launcher
+        key_path = r"Software\Python\PythonCore"
+        try:
+            import winreg
+            with winreg.OpenKey(winreg.HKEY_LOCAL_MACHINE, key_path) as key:
```

Comment:
There's two other registry locations we need to apply the same process to, to find user-land installations and 32-bit installs on 64-bit Windows.

Reply:
Thanks for your help with all this -- greatly appreciated.   Do you know what the other paths would be?  I can get back to looking at this soon.

Reply:
Per [PEP-0514](https://peps.python.org/pep-0514/#structure), the three registry paths that need to be checked are:

```
HKEY_CURRENT_USER\Software\Python\<Company>\<Tag>
HKEY_LOCAL_MACHINE\Software\Python\<Company>\<Tag>
HKEY_LOCAL_MACHINE\Software\Wow6432Node\Python\<Company>\<Tag>
```

You can check them all the same way, there's some notes there about what to do if you see the same Company-Tag combination in different registry paths, and specific notes about how to disambiguate 32-bit and 64-bit Python older than 3.5, which did not include architecture in its Tag, and so has the same Company-Tag for 32-bit and 64-bit installs.

Reply:
Just took a stab at the refactor, not really sure if it's 100% correct though.